geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/query/QueryProcess.scala

```diff
@@ -98,6 +105,9 @@ class QueryVisitor(features: SimpleFeatureCollection,
     source.getFeatures(query)
   }
 
+  override def getExpressions: java.util.List[Expression] = {
+    FilterHelper.propertyNames(filter, origSft).map(ff.property).toList
```

Comment:
need to add some extra logic here: if `properties` is null, then just return all attributes; else return properties + filter attributes (deduped)